File: src/main/java/dev/lumas/lumaitems/util/FireAnyways.java
package dev.lumas.lumaitems.util;

import dev.lumas.lumaitems.enums.Action;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
public @interface FireAnyways {
    Action[] value();
}

File: src/main/java/dev/lumas/lumaitems/annotations/FireAnyways.kt
package dev.lumas.lumaitems.annotations

import dev.lumas.lumaitems.enums.Action

/**
 * An annotation describing that a specific action should be executed for an item
 * even if the item is not being held or would not be detected normally.
 */
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS)
annotation class FireAnyways(vararg val value: Action)
